x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/InitializePolicyContextStep.java

```diff
@@ -41,6 +41,10 @@ public ClusterState performAction(Index index, ClusterState clusterState) {
 
         LifecycleExecutionState.Builder newCustomData = LifecycleExecutionState.builder(lifecycleState);
         newCustomData.setIndexCreationDate(indexMetaData.getCreationDate());
+        // The "new" phase is initialized now, usually the phase time would be
+        // set on phase transition, but since there is no transition into the
+        // "new" phase, we set it when initializing the context
+        newCustomData.setPhaseTime(System.currentTimeMillis());
```

Comment:
Elsewhere in ILM we use a `LongSupplier` that's passed in via the constructor instead of using `System.currentTimeMillis()`, should we do that here?

Reply:
I'm not sure this is the right place to set the phase time. IF we do it here we have two places in the code setting the phase/action/step time which seems a bit ugly to me. Also setting this inside a step means we can't use the `nowSupplier` and this again seems a bit ugly to me. 

I'm wondering f there is a way we can do this in `IndexLifecycleRunner.moveExecutionToNextStep()` so its in the same place that we set the times in every other case? Maybe by adding a check that if the currentPhase is `"new"` and the phase time has not been set we set it at that point?